### PR TITLE
Various refactor and features

### PR DESCRIPTION
**Pull request by vwxyzjn**
This PR brings numerous breaking changes.

1. I changed from using `argparse` to `tyro` for better code readability. 
    * plot configs are now in sub configs under `pc` (plot config)
    * `--ncols` is now `--pc.ncols`
    * `--ncols-legend` is now  `--pc.ncols-legend`
    * `--xlabel` is now `--pc.xlabel `
    * `--ylabel` is now `--pc.ylabel`
    * `--check-empty-runs False` is now `--no-check-empty-runs`
 1. I added some applicable tests in the `tests` folder for running them locally.
 2. Support for offline mode `--offline`. When we pull runs, we automatically build `sqlite` databases for storing the metadata. Then later we can run the `openrlbenchmark` in without having access to internet. 
 3. Added `rlops_hns` variant, which also integrates with `rliable` for more statistical analysis. 
image

### CODE REVIEW

**Comment by vwxyzjn** [comment]
@qgallouedec and @ffelten could you give the latest one a try? Thanks!

File: pyproject.toml [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
-version = "0.1.1b3"
+version = "0.2.0b1"
 description = ""
 authors = ["Costa Huang <costa.huang@outlook.com>"]
 readme = "README.md"
 
 [tool.poetry.dependencies]
-python = "^3.7.1"
+python = ">=3.7.1,<3.11"
 wandb = "^0.13.7"
```
Comment (ffelten) [inline pyproject.toml:10]: I think you can bump wandb

**Review by ffelten — COMMENTED** [review]
I'm getting sqlite errors: `peewee.OperationalError: unable to open database file`. Is there any setup procedure?

**Comment by vwxyzjn** [comment]
> I'm getting sqlite errors: `peewee.OperationalError: unable to open database file`. Is there any setup procedure?

What is your operating system? Also do you see  a offline.sqlite in openrlbenchmark/dataset/cleanrl folder?

**Comment by ffelten** [comment]
> > I'm getting sqlite errors: `peewee.OperationalError: unable to open database file`. Is there any setup procedure?
> 
> What is your operating system? Also do you see a offline.sqlite in openrlbenchmark/dataset/cleanrl folder?

Running on MacOS. I don't see any dataset folder. :/

**Comment by vwxyzjn** [comment]
I was able to reproduce with colab

image

**Comment by vwxyzjn** [comment]
I just pushed a fix. Could you give it another try? @ffelten

**Comment by qgallouedec** [comment]
It's a good job! Everything is working well for me so far.
At this point we have to decide a question (cc @araffin). In the future, the sb3 benchmark should contain runs with parameters that are not the default ones. For example, we might find that for an algo-env, changing the optimizer significantly increases the results. In this case, we would add the new experiments, without removing the old ones. To differentiate the first ones from the new ones, we consider adding a tag (for example, "outdated", see https://github.com/DLR-RM/rl-baselines3-zoo/issues/340#issuecomment-1380376416). What do you think about it?  Would it interface smoothly with openrlbenchmark? Have you ever considered this question for other benchmarks?

**Comment by ffelten** [comment]
@qgallouedec i think tagging is good for when we change the code, while saving config is good for hyperparameters. So you should have all you need in wandb to do such things

**Comment by qgallouedec** [comment]
Yes, but I am thinking from the perspective of openrlbenchmark. When openrlbenchmark fetches the data, how does it identify which runs should be used, and which should be ignored (side experiments, outdated parameter, etc)?

**Comment by vwxyzjn** [comment]
> Yes, but I am thinking from the perspective of openrlbenchmark. When openrlbenchmark fetches the data, how does it identify which runs should be used, and which should be ignored (side experiments, outdated parameter, etc)?

The primary way of doing it is probably through tagging, which the offline mode supports. However, I haven’t quite figured out how to “fetch the latest runs” elegantly because the latest runs might come from multiple tags.

**Comment by qgallouedec** [comment]
> I haven’t quite figured out how to “fetch the latest runs” elegantly because the latest runs might come from multiple tags.

Something like `exclude_tag` query?

It could look like

```sh
python -m openrlbenchmark.rlops \
    --filters '?we=openrlbenchmark&wpn=sb3&ceik=env&cen=algo&metric=rollout/ep_rew_mean' \
        'a2c?exclude_tag=outdated'
```

btw, I'm not sure if "outdated" is relevant, but my point is that we might want to exclude some runs that match the queried env/algo

**Comment by ffelten** [comment]
> I just pushed a fix. Could you give it another try? @ffelten

Yep, works! Tried to rerun the command as well to see if cached runs where still good. They are. 

Though I do not see the `offline.sqlite` you were mentionning in the dataset folder.

**Comment by vwxyzjn** [comment]
@ffelten could you try running the command with --offline

**Comment by ffelten** [comment]
> @ffelten could you try running the command with --offline

Works on my machine 😎

**Comment by ffelten** [comment]
> Btw the offline db is here:
> 
> ```
> 
> offline_db_folder = os.path.join( 
>          openrlbenchmark.__path__[0], "dataset", f"{wandb_entity}/{wandb_project_name}" 
>      ) 
> ```
> 
> https://github.com/openrlbenchmark/openrlbenchmark/blob/4896b8540e9d83bf55af0da5b99a3ff22c42b73c/openrlbenchmark/rlops.py#L432-L443

Yeah, I actually missed it in the middle of the folders. Sorry 🙈

**Comment by vwxyzjn** [comment]
Merging this PR now and moving discussions to #14.

> > I haven’t quite figured out how to “fetch the latest runs” elegantly because the latest runs might come from multiple tags.
> 
> Something like `exclude_tag` query?
> 
> It could look like
> 
> ```shell
> python -m openrlbenchmark.rlops \
>     --filters '?we=openrlbenchmark&wpn=sb3&ceik=env&cen=algo&metric=rollout/ep_rew_mean' \
>         'a2c?exclude_tag=outdated'
> ```
> 
> btw, I'm not sure if "outdated" is relevant, but my point is that we might want to exclude some runs that match the queried env/algo

Exclusion might be tricky because we might have hundreds of automatically generated tags. `outdated` tag is also tricky because it might involve updating the tags that are in wandb and the offline db, which could cause breaking changes (e.g., the runs tagged with `outdated` today might be different tomorrow). Let's have further discussions at #14.

**Pull request merged**